### PR TITLE
Add more logs into OpenShift failure cases

### PR DESCRIPTION
**Pull request by phracek**
This pull request adds more logs in case RHSCL image builds are failing.

E.g. When we are building s2i images, we do not know reason, why it failed.

Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

### CODE REVIEW

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

File: test-lib-openshift.sh [hunk 1]
```diff
@@ -169,6 +169,21 @@ function ct_os_get_pod_ip() {
...
 
+# ct_os_get_sti_build_logs
+# -----------------
+# Return logs from sti_build
+# Arguments: pod_name
+function ct_os_get_sti_build_logs() {
+  local pod_name="${1}"
+  # Print logs but do not failed. Just for traces
+  if oc get "bc/$pod_name"; then
+    oc logs "bc/$pod_name" || return 0
```
Comment (pkubatrh) [inline test-lib-openshift.sh:180]: Will this work? In all uses of the function `pod_prefix` is passed as the `pod_name` argument. Looking at how the other functions use `pod_prefix` (grep is involved) it does not seem like `oc` would be able to work with just the prefix.

File: test-lib-openshift.sh [hunk 2]
```diff
@@ -195,7 +210,12 @@ function ct_os_wait_pod_ready() {
     echo -n "Waiting for ${pod_prefix} build pod to finish ..."
     while ! [ "$(ct_os_get_build_pod_status "${pod_prefix}")" == "Succeeded" ] ; do
       echo -n "."
-      [ "${SECONDS}" -gt "${timeout}0" ] && echo " FAIL" && return 1
+      [ "${SECONDS}" -gt "${timeout}0" ] && {
+        echo " FAIL";
+        ct_os_print_logs || :
+        ct_os_get_sti_build_logs "${pod_prefix}"
+        return 1
+      }
```
Comment (pkubatrh) [inline test-lib-openshift.sh:218]: I would rewrite it as an `if` now that it spans multiple lines to make it more readable. Same for the ones below

Comment (pkubatrh) [reply test-lib-openshift.sh:218]:
```suggestion
      if [ "${SECONDS}" -gt "${timeout}0" ]; then
        echo " FAIL";
        ct_os_print_logs || :
        ct_os_get_sti_build_logs "${pod_prefix}"
        return 1
      fi
```

File: tests/squash/squash.sh [hunk 3]
```diff
@@ -8,7 +8,7 @@ if grep -q "Red Hat Enterprise Linux release 8" /etc/system-release; then
   exit 0
 fi
 
-origin=busybox
+origin=quay.io/centos7/s2i-core-centos7
```
Comment (pkubatrh) [inline tests/squash/squash.sh:11]: Do we need to use s2i-core here? I would use something small if we do not need any of s2i-core's functionality

Comment (phracek) [reply tests/squash/squash.sh:11]: ok, It should be from quay.io or registry.redhat.io. Not from docker hub. May by ubi7/ubi.

I will test it.

File: test-lib-openshift.sh (hunk 1 above)
Comment (phracek) [reply test-lib-openshift.sh:180]: Well, I have tested it on s2i-python-container.
pod_name here is something like `pod/python-testing-1-596hh` but buildconfig, shortly `bc` is python-testing.

See log from OpenShift:
```bash
Waiting for python-testing pod becoming ready ....................................................................................... FAIL
NAME                          READY   STATUS      RESTARTS   AGE
pod/python-testing-1-596hh    0/1     Running     3          5m6s
pod/python-testing-1-build    0/1     Completed   0          6m24s
pod/python-testing-1-deploy   1/1     Running     0          5m45s

[..snip..]

NAME                                                REVISION   DESIRED   CURRENT   TRIGGERED BY
deploymentconfig.apps.openshift.io/postgresql       0          1         0         config,image(postgresql:10)
deploymentconfig.apps.openshift.io/python-testing   1          1         1         config,image(python-testing:latest)

NAME                                            TYPE     FROM         LATEST
buildconfig.build.openshift.io/python-testing   Source   Git@master   1

NAME                                        TYPE     FROM          STATUS     STARTED         DURATION
build.build.openshift.io/python-testing-1   Source   Git@7cbc596   Complete   5 minutes ago   1m25s

[..snip..]

svc/postgresql - 172.30.227.114:5432
  dc/postgresql deploys istag/postgresql:10
    deployment #1 waiting on image or update

http://python-testing-sclorg-test-7600.apps.core-serv-ocp.hosted.psi.rdu2.redhat.com (svc/python-testing)
  dc/python-testing deploys istag/python-testing:latest <-
    bc/python-testing source builds https://github.com/sclorg/django-ex.git#master on istag/python:2.7
```

But we can detect buildconfig in a function like here: https://github.com/sclorg/container-common-scripts/blob/master/test-lib-openshift.sh#L136

`oc get bc -o "custom-columns=NAME:.metadata.name,Ready:status.phase" | grep "${pod_prefix}" | sort -u | tail -n 1`

**Comment by phracek** [comment]
[test]

File: tests/squash/squash.sh (hunk 3 above)
Comment (phracek) [reply tests/squash/squash.sh]: @pkubatrh registry.access.redhat.com/ubi8-micro does not make sense, I think. For RHEL8, squash is ignored https://github.com/sclorg/container-common-scripts/blob/master/tests/squash/squash.sh#L7.

But what about using registry.access.redhat.com/ubi7/ubi? ubi8-micro is not released yet;)

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
rebased

[test]

**Comment by phracek** [comment]
centos7 fails here:
```bash
11:25:09 open /etc/docker/certs.d/registry.access.redhat.com/redhat-ca.crt: no such file or directory
11:25:09 
11:25:09 Traceback (most recent call last):
11:25:09   File "/root/sources/tests/squash/../../squash.py", line 22, in <module>
11:25:09     from_layer=sys.argv[2]).run()
11:25:09   File "/usr/local/python-tools/lib/python3.6/site-packages/docker_squash/squash.py", line 61, in run
11:25:09     return self.squash(image)
11:25:09   File "/usr/local/python-tools/lib/python3.6/site-packages/docker_squash/squash.py", line 91, in squash
11:25:09     new_image_id=image.squash()
11:25:09   File "/usr/local/python-tools/lib/python3.6/site-packages/docker_squash/image.py", line 73, in squash
11:25:09     self._before_squashing()
11:25:09   File "/usr/local/python-tools/lib/python3.6/site-packages/docker_squash/v2_image.py", line 15, in _before_squashing
11:25:09     super(V2Image, self)._before_squashing()
11:25:09   File "/usr/local/python-tools/lib/python3.6/site-packages/docker_squash/image.py", line 153, in _before_squashing
11:25:09     self.old_image_id = self.docker.inspect_image(self.image)['Id']
11:25:09   File "/usr/local/python-tools/lib/python3.6/site-packages/docker/utils/decorators.py", line 17, in wrapped
11:25:09     'Resource ID was not provided'
11:25:09 docker.errors.NullResource: Resource ID was not provided
```

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
@pkubatrh I think, we can not use registry.access.redhat.com/ubi8-micro because CentOS ci does not have Red Hat certificate. Therefore I would prefer to use basic s2i-core from quay.io.

What do you think?

But I will contact CI folks, whether there is some package that needs to be installed, which provides the certificate file.

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
@pkubatrh Tests are passing. As soon as it is LGTM. we can merge it.

**Comment by phracek** [comment]
This blocks PR like https://github.com/sclorg/container-common-scripts/pull/199
Tests are PASSING.

Merging.

**Pull request merged**